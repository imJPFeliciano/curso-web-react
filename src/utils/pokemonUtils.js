// estandariza el identificador del pokemon a 3 digitos
export const sanitizePokemonId = pokemonId => {
  let sanitizedId;

  if (pokemonId < 10) sanitizedId = `00${pokemonId}`;
  else if (pokemonId < 100) sanitizedId = `0${pokemonId}`;
  else sanitizedId = pokemonId;

  return sanitizedId;
}

export const getPokemonImage = pokemonId => {
  return `https://assets.pokemon.com/assets/cms2/img/pokedex/detail/${pokemonId}.png`;
}
