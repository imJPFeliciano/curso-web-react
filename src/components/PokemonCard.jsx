import React from 'react';
import { sanitizePokemonId, getPokemonImage } from '../utils/pokemonUtils';
import { PokemonCardContainer, PokemonName, PokemonImage } from './Pokemon.styled';

const PokemonCard = (props) => {
  const { name, url } = props;
  let pokemonId = Number(url.split('/')[6]);
  pokemonId = sanitizePokemonId(pokemonId); // 3 digitos

  return (
    <PokemonCardContainer>
      <PokemonImage src={getPokemonImage(pokemonId)} />
      <PokemonName>{pokemonId}: {name}</PokemonName>
    </PokemonCardContainer>
  );
};

export default PokemonCard;
