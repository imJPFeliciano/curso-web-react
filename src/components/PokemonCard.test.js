import React from 'react';
import PokemonCard from './PokemonCard';
import { mount, shallow } from 'enzyme';
import { sanitizePokemonId } from '../utils/pokemonUtils';
import { PokemonName, PokemonImage } from './Pokemon.styled';

const name = 'Bulbasaur';
const url = 'https://pokeapi.co/api/v2/pokemon/1';

let pokemonId = Number(url.split('/')[6]);
pokemonId = sanitizePokemonId(pokemonId);

describe('Pokemon Card', () => {
  it('render sin problemas', () => {
    const wrapper = shallow(<PokemonCard name={name} url={url} />);
    expect(wrapper).toMatchSnapshot();  
  });

  it('muestra el componente de nombre', () => {
    const wrapper = shallow(<PokemonCard name={name} url={url} />);
    expect(wrapper.find(PokemonName)).toHaveLength(1);
  });

  it('muestra el componente de imagen', () => {
    const wrapper = shallow(<PokemonCard name={name} url={url} />);
    expect(wrapper.find(PokemonImage)).toHaveLength(1);
  });
})