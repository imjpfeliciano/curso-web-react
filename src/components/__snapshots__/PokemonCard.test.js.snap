// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Card render sin problemas 1`] = `ShallowWrapper {}`;
